android/src/main/java/com/amplitude/android/utilities/Session.kt

```diff
@@ -0,0 +1,130 @@
+package com.amplitude.android.utilities
+
+import com.amplitude.android.Amplitude
+import com.amplitude.android.Configuration
+import com.amplitude.core.State
+import com.amplitude.core.Storage
+import com.amplitude.core.events.BaseEvent
+import java.util.concurrent.atomic.AtomicLong
+
+class Session(
+    private var configuration: Configuration,
+    private var storage: Storage? = null,
+    private var state: State? = null,
+) {
+    companion object {
+        const val EMPTY_SESSION_ID = -1L
+        const val DEFAULT_LAST_EVENT_TIME = -1L
+        const val DEFAULT_LAST_EVENT_ID = 0L
+    }
+
+    private val _sessionId = AtomicLong(EMPTY_SESSION_ID)
+    private val _lastEventId = AtomicLong(DEFAULT_LAST_EVENT_ID)
```

Comment:
Curious - what is the relationship between last event id and a session?

Reply:
I felt this made sense to be part of the session since it is tied to the the events sent during the session. wdyt?

Reply:
It does not get reset at the end of a session though right? The reason I brought this up was because another pet peeve I had was that identify suppression also makes our event ids to be skipped. Was hoping to have event id increment happen after identify suppression. 

Not a blocker. We can tackle those problems separately.

Reply:
Yeah, I agree, it's a little weird because it is shared across sessions. Originally I was thinking we would reset on a new session.